js/classes/entities/midground/enemies/overseer_bot.js

```diff
@@ -0,0 +1,228 @@
+    static get ATTACK_DAMAGE() {
+        return 10;
+    };
+
+    /** Number of seconds after the start of the attack animation when damage should be dealt. */
+    static get DAMAGE_DELAY() {
+        return 0.4;
+    }
+
+    /**
+     * Perform any necessary operations when the OverseerBot dies.
+     */
+    handleDeath() {
+        this.action = "dying";
+
+        GAME.addEntity(new ParticleEffect(this.getCenter(), ParticleEffect.GRAY_SPARKLE));
+
+        const rand = Math.random();
+        if (rand < 0.33) {
+            ASSET_MGR.playSFX(SFX.ROBOT_DEATH1.path, SFX.ROBOT_DEATH1.volume);
+        } else if (rand < 0.66) {
+            ASSET_MGR.playSFX(SFX.ROBOT_DEATH2.path, SFX.ROBOT_DEATH2.volume);
+        } else {
+            ASSET_MGR.playSFX(SFX.ROBOT_DEATH3.path, SFX.ROBOT_DEATH3.volume);
+        }
+
+        const pos = Vector.add(this.getCenter(), new Vector(0, -80));
+        // add a piece of food in the bot's place at bottom-center of bot
+        if (Math.random() < 0.3) {
+            GAME.addEntity(new FoodDrop(pos, FoodDrop.ENERGY_DRINK));
+        }
+
+        if (Math.random() < 0.5) {
+            GAME.addEntity(new AmmoDrop(pos, AmmoDrop.BOMB, 3));
+        }
+
+        if (Math.random() < 0.5) {
+            GAME.addEntity(new RuneDrop(pos, RuneDrop.GREEN));
+        } else {
+            GAME.addEntity(new RuneDrop(pos, RuneDrop.RED));
+        }
+    }
+    
+    /** Change what the OverseerBot is doing and where it is. */
+    update() {
+        this.base.update();
+
+        const deathAnim = this.animations[this.getFacing()]["dying"];
+
+        if (this.health > 0) {
+            const secondsSinceLastAttack = Date.now() / 1000 - this.lastAttack;
+
+            // if we've finished our current attack, change action to idle
+            if (this.action === "attacking" 
+                && this.animations[this.getFacing()]["attacking"].totalTime < secondsSinceLastAttack) {
+                    
+                this.action = "moving";
+            }
+    
+            // if Chad is close enough, shoot him
+            if (Vector.distance(CHAD.pos, this.pos) < OverseerBot.ATTACK_DISTANCE) {
+                if (secondsSinceLastAttack > OverseerBot.ATTACK_COOLDOWN) {
+                    // if it's been long enough, start a new attack 
+                    this.animations[this.getFacing()]["attacking"].elapsedTime = 0;
+                    this.action = "attacking";
+                    this.lastAttack = Date.now() / 1000;
+                    this.dealtDamage = false;
+                } else if (this.action === "attacking" 
+                    && secondsSinceLastAttack > OverseerBot.DAMAGE_DELAY && !this.dealtDamage) {
+                    // if we're at the proper point in our attack animation, deal damage
+    
+                    const misdirectVector = new Vector(Math.random() * 20 - 10, Math.random() * 20 - 10);
+                    const target = Vector.add(CHAD.getCenter(), misdirectVector);
```

Comment:
What is misdirectVector?

Reply:
It's used to slightly change the direction of the enemy's projectile. It's like stormtrooper aiming. I thought it was a nice touch, but let me know if it's unnecessary.

Reply:
No this is great. I love the idea. The naming was a bit confusing but makes more sense now. It might be funny and clear to name it something like "stormtrooperAim" but either way.